src/nuklear_draw.c

```diff
@@ -556,3 +556,397 @@ nk_draw_text(struct nk_command_buffer *b, struct nk_rect r,
     cmd->string[length] = '\0';
 }
 
+#ifdef NK_ENABLE_SUBPIXEL_API
+
+NK_API void
+nk_push_scissor_subpixel(struct nk_command_buffer *b, struct nk_rect r)
+{
+    struct nk_command_scissor *cmd;
+    NK_ASSERT(b);
+    if (!b) return;
+
+    b->clip.x = r.x;
+    b->clip.y = r.y;
+    b->clip.w = r.w;
+    b->clip.h = r.h;
+    cmd = (struct nk_command_scissor*)
+        nk_command_buffer_push(b, NK_COMMAND_SCISSOR, sizeof(*cmd));
+
+    if (!cmd) return;
+    cmd->x = r.x;
+    cmd->y = r.y;
+    cmd->w = NK_MAX(0, r.w);
+    cmd->h = NK_MAX(0, r.h);
+}
```

Comment:
There are a lot of these new subpixel apis that are just copies of the original but without the casts to `short`. Their signature are the same, so should `nk_push_scissor` just support subpixels if enabled instead of having two functions?

Reply:
Not sure if this was a bad auto merge or for backwards compatibility.

Reply:
>so should `nk_push_scissor` just support subpixels if enabled instead of having two functions?

Interesting idea. I think it's worth exploring. @Nielsbishere would you find some time to try to rewrite this PR to see if it solves some of the issues we have here and if it makes the PR leaner?